src/main/kotlin/CurrencyConversion.kt

```diff
@@ -0,0 +1,106 @@
+import java.text.NumberFormat
+import java.util.*
+
+class CurrencyConversion {
+
+    public fun currencyConverse(userInput: String): String {
+        val infosInput = splitCurrency(userInput);
+
+        if (!codeIsValid(infosInput.target.name) && !codeIsValid(infosInput.source.name)) {
+            return "ERRO! Taxas invalidas"
+        }
+        if (!amountIsValid(infosInput.amountToConversion)) {
+            return "ERRO! Valor Negativo"
+        }
+        val result = conversion(infosInput);
+        return ShowResult(result)
+    }
+
+    //matriz com as taxas de cambio
+    private val currencyExchengeRate = arrayOf(    //AUD       BRL      EUR       IRR         USD
+                                            arrayOf(1.0,     4.3009,  0.6403,  32631.3775,  0.7630),   //AUD
+                                            arrayOf(0.2324,  1.0,     0.1491,  7493.7254,   0.1774),   //BRL
+                                            arrayOf(1.5616,  6.7170,  1.0,     50403.8615,  1.5615),   //EUR
+                                            arrayOf(0.0000,  0.0001,  0.0000,  1.0,         0.0000),   //IRR
+                                            arrayOf(1.31021, 5.6360,  0.8380,  42105.0068,    1.0))    //USD
+
+    //possiveis moedas de conversão
+    private enum class CurrencyCode { AUD, BRL, EUR, IRR, USD }
+
+    //Classe que contém os valores para conversão
+    private data class Conversion(
+        val source: CurrencyCode,
+        val target: CurrencyCode,
+        val amountToConversion: Double,
+        var rate: Double = 0.0,
+        var amountConverted: Double = 0.0
+    )
+
+    // verificar se os codes estão listados no sistema
+    private fun codeIsValid(code: String): Boolean = code in CurrencyCode.values().map(CurrencyCode::name)
+
+    // verificar se o valor para conversão é valido
+    private fun amountIsValid(amount: Double): Boolean = amount >= 0
```

Comment:
Acho que o nome dessa função já está bom, mas quanto mais óbvio for o nome melhor, queria deixar como sugestão: `amountIsPositive`